odml/property.py

```diff
@@ -102,7 +102,7 @@ def parent(self):
     @parent.setter
     def parent(self, new_parent):
         if new_parent is None:
-            return
+            self._parent = None
         elif self._validate_parent(new_parent):
```

Comment:
I think this will fail :) 
We still need to return after setting it to None, most likely the following checks will fail.